extras/ds18b20/ds18b20.c

```diff
@@ -0,0 +1,111 @@
+#include "onewire/onewire.h"
+#include "ds18b20.h"
+
+#define DS1820_WRITE_SCRATCHPAD 0x4E
+#define DS1820_READ_SCRATCHPAD  0xBE
+#define DS1820_COPY_SCRATCHPAD  0x48
+#define DS1820_READ_EEPROM      0xB8
+#define DS1820_READ_PWRSUPPLY   0xB4
+#define DS1820_SEARCHROM        0xF0
+#define DS1820_SKIP_ROM         0xCC
+#define DS1820_READROM          0x33
+#define DS1820_MATCHROM         0x55
+#define DS1820_ALARMSEARCH      0xEC
+#define DS1820_CONVERT_T        0x44
+
+uint8_t readDS18B20(uint8_t pin, DSENSOR *result){
+    static uint8_t one_time = 1;
+    
+    if (one_time){
+        onewire_init(pin);
```

Comment:
Rather than lazy-initialising on demand, could you expose `onewire_init` in the header file and encourage explicit initialisation as part of the API?